From 5ae46a1d36dadb64696e4d16516db1364891ae55 Mon Sep 17 00:00:00 2001
Message-Id: <5ae46a1d36dadb64696e4d16516db1364891ae55.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Wed, 20 Aug 2014 16:12:40 +0530
Subject: [PATCH 02/75] MUST_REBASE [VPG]: adf/Kconfig: Add support for INTEL
 ADF driver.

v2: Fixed compilation issue with CONFIG_ADF_FBDEV

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I8ea32d3981c0aedfded37814fa9b0b07e68767ea
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/video/adf/Kconfig        |   28 +++++++++++++++++++++++-----
 drivers/video/adf/Makefile       |    2 ++
 drivers/video/adf/intel/Makefile |    2 ++
 3 files changed, 27 insertions(+), 5 deletions(-)

diff --git a/drivers/video/adf/Kconfig b/drivers/video/adf/Kconfig
index 33858b7..3563f6a 100644
--- a/drivers/video/adf/Kconfig
+++ b/drivers/video/adf/Kconfig
@@ -1,14 +1,32 @@
 menuconfig ADF
 	depends on SYNC
 	depends on DMA_SHARED_BUFFER
-	tristate "Atomic Display Framework"
+	tristate "Atomic Display Framework (ADF)"
+	help
+	  Enable support for Atomic display framework.
 
-menuconfig ADF_FBDEV
+config ADF_FBDEV
 	depends on ADF
 	depends on FB
-	tristate "Helper for implementing the fbdev API in ADF drivers"
+	tristate "ADF fbdev helper"
+	help
+	  Enable ADF framebuffer driver helpere.
 
-menuconfig ADF_MEMBLOCK
+config ADF_MEMBLOCK
 	depends on ADF
 	depends on HAVE_MEMBLOCK
-	tristate "Helper for using memblocks as buffers in ADF drivers"
+	tristate "ADF memblock helpers"
+	help
+	  Enable ADF memory block helpers
+
+config ADF_INTEL
+	depends on ADF
+	tristate "Intel Display Driver"
+	help
+	  Enable ADF compatible Intel display driver.
+
+config ADF_INTEL_FBDEV
+	depends on ADF_INTEL && ADF_FBDEV
+	tristate "Intel FBDEV Driver "
+	help
+	  Enable ADF based framebuffer driver for Intel displays.
diff --git a/drivers/video/adf/Makefile b/drivers/video/adf/Makefile
index d5604d1..1d11737 100644
--- a/drivers/video/adf/Makefile
+++ b/drivers/video/adf/Makefile
@@ -13,3 +13,5 @@ obj-$(CONFIG_COMPAT) += adf_fops32.o
 obj-$(CONFIG_ADF_FBDEV) += adf_fbdev.o
 
 obj-$(CONFIG_ADF_MEMBLOCK) += adf_memblock.o
+
+obj-$(CONFIG_ADF_INTEL)  += intel/
diff --git a/drivers/video/adf/intel/Makefile b/drivers/video/adf/intel/Makefile
index e55da6e..4c7cf75 100644
--- a/drivers/video/adf/intel/Makefile
+++ b/drivers/video/adf/intel/Makefile
@@ -11,4 +11,6 @@ obj-$(CONFIG_ADF_INTEL_FBDEV) += intel_adf_fbdev.o
 
 obj-$(CONFIG_ADF_MEMBLOCK) += intel_adf_mm.o
 
+obj-$(CONFIG_ADF_INTEL_FBDEV) += intel_adf_fbdev.o
+
 obj-$(CONFIG_ADF_INTEL) += core/
-- 
1.7.9.5

